### PR TITLE
First mirrors optimization pass

### PR DESCRIPTION
**Pull request by Geenz**
Adds a ton of stuff for conservative updates on mirrors, including per face conservative updates and the like that scale based upon face visibility.  I think the face layout I have may be wrong - but overall all of this combined boosts the framerate on my systems to be well within spec (50% impact on framerate).

### CODE REVIEW

**Comment by Geenz** [comment]
Also noticed - this brings us slightly more up to date with main.  I must have accidentally merged up with main in the last couple of weeks - my bad!

File: indra/newview/pipeline.cpp [hunk 1]
```diff
@@ -1068,6 +1075,9 @@ void LLPipeline::refreshCachedSettings()
         LLViewerShaderMgr::instance()->clearShaderCache();
         LLViewerShaderMgr::instance()->setShaders();
     }
+    RenderHeroProbeUpdateRate = gSavedSettings.getS32("RenderHeroProbeUpdateRate");
+    RenderHeroProbeConservativeUpdateMultiplier = gSavedSettings.getS32("RenderHeroProbeConservativeUpdateMultiplier");
+
```
Comment (RunitaiLinden) [inline indra/newview/pipeline.cpp:1080]: Prefer to not use connectRefreshCachedSettings and statics, but use static LLCachedControl.

**Review by RunitaiLinden — APPROVED** [review]
Approved with comment... not a blocker, but probably refrain from using connectRefreshCachedSettings and instead use static LLCachedControl.

**Review by brad-linden — APPROVED** [review]

File: indra/newview/pipeline.cpp [hunk 2]
```diff
@@ -786,9 +790,12 @@ bool LLPipeline::allocateScreenBuffer(U32 resX, U32 resY, U32 samples)
...
         allocateScreenBuffer(res, res, samples);
 
-		res = mHeroProbeManager.mProbeResolution; // We also scale the hero probe RT to the probe res since we don't super sample it.
-        mRT = &mHeroProbeRT;
-        allocateScreenBuffer(res, res, samples);
+		if (RenderMirrors)
+        {
+            res = mHeroProbeManager.mProbeResolution;  // We also scale the hero probe RT to the probe res since we don't super sample it.
+            mRT = &mHeroProbeRT;
+            allocateScreenBuffer(res, res, samples);
```
Comment (cosmic-linden) [inline indra/newview/pipeline.cpp:797]: Not a blocker but I notice that the hero probe target isn't released when RenderMirrors is set to false.

**Review by cosmic-linden — APPROVED** [review]

**Pull request merged**